src/snowflake/snowpark/mock/plan.py

```diff
@@ -438,6 +437,17 @@ def execute_mock_plan(
                         infer_type(cal_exp_res), nullable=True
                     )
             data.append(values)
+
+        if not children_dfs.indices:
+            aggregate_by_groups(child_rf)
+        else:
+            for _, indices in children_dfs.indices.items():
```

Comment:
Unrelated to your change: Can't we do `values()`?

Reply:
We could!